src/debug.c

```diff
@@ -285,34 +285,32 @@ void computeDatasetDigest(unsigned char *final) {
 
     for (j = 0; j < server.dbnum; j++) {
         redisDb *db = server.db+j;
+        if (dbSize(db) == 0) continue;
+        dbIterator dbit;
+        dbIteratorInit(&dbit, db);
 
-        if (dictSize(db->dict) == 0) continue;
-        di = dictGetSafeIterator(db->dict);
-
-        /* hash the DB id, so the same dataset moved in a different
-         * DB will lead to a different digest */
+        /* hash the DB id, so the same dataset moved in a different DB will lead to a different digest */
         aux = htonl(j);
-        mixDigest(final,&aux,sizeof(aux));
+        mixDigest(final, &aux, sizeof(aux));
 
         /* Iterate this DB writing every entry */
-        while((de = dictNext(di)) != NULL) {
+        while ((de = dbIteratorNext(&dbit)) != NULL) {
             sds key;
             robj *keyobj, *o;
 
-            memset(digest,0,20); /* This key-val digest */
+            memset(digest, 0, 20); /* This key-val digest */
             key = dictGetKey(de);
```

Comment:
Here too, nice to have unchanged indentation, but some extra whitespace changes can be reverted (not only these but some more).